### ISSUE TITLE
getFileContent response returns empty object when it's a PDF file (JPG files works fine)

### ISSUE THREAD

**Issue by estacioneto** [Alfresco/alfresco-js-api#271]
**Type of issue:**  (check with "[x]")
```
- [ ] New feature request
- [X ] Bug  
- [ ] Support request
```
**Current behavior:**
Response from `getFileContent` of a PDF always retruns empty object.

**Expected behavior:**
Response from `getFileContent` returning Buffer (as it happens with JPG files).

**Steps to reproduce the issue:**
* Upload a PDF file;
* Call `getFileContent` from a NodeJS server;
* The response will be an empty object instead of a Buffer.

**Node version (for build issues):**  
v8.4.0

**New feature request:**

**Comment by DenisVuyka** [comment]
@estacioneto What is the version of the library and ACS you are using? We got a lot of testing for PDFs and the retrieval has always been working fine.

**Comment by estacioneto** [comment]
@DenisVuyka The library's version is 1.8.0 and the Alfresco's version is 5.2. It's really strange, because `getNodeInfo` works well (so the id of the file is correct).

**Comment by estacioneto** [comment]
`const file = await alfrescoApi.core.nodesApi.getFileContent('6ec51a5c-23e1-4818-9d2f-8baef787e628');
console.log('file', file, _.isEmpty(file)); // file {} true
const node = await alfrescoApi.nodes.getNodeInfo('6ec51a5c-23e1-4818-9d2f-8baef787e628');
console.log('node', node, _.isEmpty(node)); // prints correct informations`

**Comment by estacioneto** [comment]
@DenisVuyka Any ideas? Am I using it incorrectly?

**Comment by DenisVuyka** [comment]
From the ADF this API works fine as I've mentioned:

```ts
this.apiService.getInstance().nodes
                .getFileContent(node.id)
                .then(response => {
                    console.log(response);
                });
```

Can you check that the call is authenticated, and probably another check without async/await?

**Comment by DenisVuyka** [comment]
Also, I suggest trying `alfrescoApi.nodes.getFileContent`

**Comment by estacioneto** [comment]
@DenisVuyka The strange thing happening here is that I can get an image content but not PDF (I verified the authentication). I also tried other API functions (`getContentUrl`, `getFileInfo`) and worked fine.

```javascript
alfrescoApi.nodes.getFileContent('fa467304-dcbf-420b-8f19-95081e7ffb03')
    .then(response => {
        console.log('image (jpg)', response); // image (jpg) <Buffer 89 50 4e 47 0d 0a 1a ... >
    });

alfrescoApi.nodes.getFileContent('d7c015f8-73a7-4876-8612-404872f3d47b')
    .then(response => {
        console.log('document (pdf)', response); // document (pdf) {}
    });
```

**Comment by DenisVuyka** [comment]
What version of the repository are you using?

**Comment by estacioneto** [comment]
Do you mean this version?
![captura de tela de 2017-09-19 11-55-24](https://user-images.githubusercontent.com/15948386/30598945-7756c4d8-9d31-11e7-97b6-5ef002fba561.png)

**Comment by DenisVuyka** [comment]
I can't reproduce the issue locally, so will need more details I'm afraid. Are you sure you are fetching the real PDF file content and not the rendition to PDF from some other format?

**Comment by estacioneto** [comment]
@DenisVuyka This is the output from `getNodeInfo`:

```
document (pdf) { isFile: true,
  createdByUser: { id: 'admin', displayName: 'Administrator' },
  modifiedAt: 2017-09-19T13:56:52.481Z,
  nodeType: 'cm:content',
  content: 
   { mimeType: 'application/pdf',
     mimeTypeName: 'Adobe PDF Document',
     sizeInBytes: 60431,
     encoding: 'UTF-8' },
  parentId: '803da866-86c4-4aab-90f6-6eee714da4d7',
  aspectNames: 
   [ 'rn:renditioned',
     'cm:versionable',
     'cm:titled',
     'cm:auditable',
     'cm:taggable',
     'cm:author',
     'cm:thumbnailModification' ],
  createdAt: 2017-09-19T12:55:45.985Z,
  isFolder: false,
  modifiedByUser: { id: 'admin', displayName: 'Administrator' },
  name: 'anexo_1505825745881.pdf',
  id: 'd7c015f8-73a7-4876-8612-404872f3d47b',
  properties: 
   { 'cm:title': 'þÿ',
     'cm:versionType': 'MAJOR',
     'cm:versionLabel': '1.0',
     'cm:lastThumbnailModification': 'doclib:1505825746367' } }
```

I hope it helps! You can see the id is the same of the latest comment with contents.

**Comment by montgolfiere** [comment]
Does it work directly via REST API, using same authenticated user ? 

https://api-explorer.alfresco.com/api-explorer/#!/nodes/getNodeContent
=> 

eg. via Postman or curl ...

http://localhost:8080/alfresco/api/-default-/public/alfresco/versions/1/nodes/d7c015f8-73a7-4876-8612-404872f3d47b/content

**Comment by estacioneto** [comment]
@montgolfiere I tested via Postman and it worked (without any request headers, by the way)! Why? Any idea why it doesn't work via NodeJS?

**Comment by estacioneto** [comment]
I'm still with the problem. This code works:

```javascript
const url = alfrescoApi.content.getContentUrl(id);
return request({
  url,
  encoding: null
});
```

But the API call for the content still doesn't work.

**Comment by eromano** [comment]
@estacioneto are you able to provide an example project where test this problem? with some description on how to reproduce it?

**Comment by DenisVuyka** [comment]
Another option is to make a regular http call from the node.js

**Comment by estacioneto** [comment]
@eromano @DenisVuyka There is a sample project to show the problem: https://github.com/estacioneto/alfresco-js-api-file-content

**Comment by DenisVuyka** [comment]
@estacioneto I can confirm the issue, thanks for the project to reproduce. This will be fixed in the next version. Meanwhile, you can use "superagent" library directly to fetch the content in the node.js runtime, for example:

```js
const request = require('superagent');

request
        .get('http://localhost:3000/ecm/alfresco/api/-default-/public/alfresco/versions/1/nodes/46183fd9-e7c2-4598-8b53-10c4d54638ce/content')
        .auth(USERNAME, PASSWORD)
        .buffer(true)
        .end((err, res) => {
            if (err) {
                console.log(err);
                return;
            }
            console.log(res.body);
        });
```

will require installing an extra library:

```sh
yarn add superagent
```

**Comment by estacioneto** [comment]
@DenisVuyka I'll do it. Thank you :)

**Comment by DenisVuyka** [comment]
The issue is fixed and will be available in the next release or upcoming betas

**Comment by estacioneto** [comment]
@DenisVuyka any update?

**Comment by DenisVuyka** [comment]
@estacioneto see my previous comment from 13 Oct, the issue is fixed and can be available in the beta or next release.

**Comment by estacioneto** [comment]
@DenisVuyka I didn't see it's available in 1.10.0 betas. My bad. Thanks
